controllers/healthcheck/machine.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate counterfeiter . Client
+type Client interface {
+	Delete(context.Context, runtime.Object, ...client.DeleteOption) error
+}
+
+func NewUnhealthyMachineDeletionRemediator(client Client, logger logr.Logger) *UnhealthyMachineDeletionRemediator {
+	return &UnhealthyMachineDeletionRemediator{
+		client: client,
+		logger: logger,
+	}
+}
+
+type UnhealthyMachineDeletionRemediator struct {
+	client Client
+	logger logr.Logger
+}
+
+func (r *UnhealthyMachineDeletionRemediator) Remediate(ctx context.Context, machines []clusterv1.Machine) (bool, error) {
+	var (
+		errs    []error
+		deleted bool
+	)
+
+	for i := range machines {
+		machine := machines[i]
+		if _, unhealthy := machine.GetAnnotations()[clusterv1.MachineUnhealthy]; !unhealthy {
+			continue
+		}
+
+		r.logger.Info("Deleting unhealthy machine", "machine", machine.GetName())
+		if err := r.client.Delete(ctx, &machine); err != nil {
+			errs = append(errs, err)
+		}
+
+		deleted = true
+	}
+
+	return deleted, kerrors.NewAggregate(errs)
```

Comment:
Should we instead return the number of deleted Machines?

Reply:
We could, but all we care about is whether we deleted any or none as far as I can think of

Reply:
I can see that some clients might want to have the number of successfully deleted machines and compare it to the input